src/java/org/apache/cassandra/index/sai/disk/io/RAMIndexOutput.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk.io;
+
+import java.io.IOException;
+
+import org.apache.lucene.store.GrowableByteArrayDataOutput;
+import org.apache.lucene.store.IndexOutput;
+
+// Use in place of RAMOutputStream (which has monitor locks)
+public class RAMIndexOutput extends IndexOutput
+{
+    protected final GrowableByteArrayDataOutput out;
+
+    public RAMIndexOutput(String name)
+    {
+        super("", name);
+        out = new GrowableByteArrayDataOutput(128);
```

Comment:
nit: Might be a good idea to explain our reasoning for 128, even if it's not configurable.

Reply:
Well I honestly don't know how / why we ended up at that value. Looking at it now, 128 seems quite small to be starting with because the `GrowableByteArrayDataOutput` only seems to grow by how much is being added. 

If it's ok with you I'd like to raise a separate ticket to investigate this value and see if changing it's initial size has impact on performance and/or GC. I can reference that ticket in a comment.